src/main/java/com/thirdcc/webapp/web/rest/CourseProgramResource.java

```diff
@@ -50,4 +50,18 @@ public CourseProgramResource(CourseProgramService courseProgramService) {
         HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(uriBuilder.queryParams(queryParams), page);
         return ResponseEntity.ok().headers(headers).body(page.getContent());
     }
+
+    /**
+     * {@code GET  /course-programs/faculty/{facultyId}} : get all the course program.
+     *
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)} and the list of faculties in body.
+     */
+    @GetMapping("/course-programs/faculty")
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity<List<CourseProgram>> getAllCoursePrograms() {
+        log.debug("REST request to get all Course Programs");
+        List<CourseProgram> entityList = courseProgramService.findAll();
+        return ResponseEntity.ok().body(entityList);
+    }
```

Comment:
why your java doc and the API URL is not tally?
implementation is doing get all
but java doc seems like you want to find based on faculty id

Reply:
Resolved in 3d3667fc1d901fb47691745754896ff1ea423842